azimuth/modules/dataset_analysis/dataset_warnings.py

```diff
@@ -114,6 +124,41 @@ def get_general_warnings(
                     train_dist,
                     eval_dist,
                     min_num_per_class,
+                    alert_train_nb_min,
+                    alert_eval_nb_min,
+                    class_names,
+                ),
+            ),
+            DatasetWarning(
+                name=f"Class imbalance" f" (>{max_delta_imb*100:.0f}%)",
+                description=f"Relative difference between the number of "
+                f"samples per class and the mean in each dataset split is above "
+                f"{max_delta_imb*100:.0f}%.",
+                columns=["training", "evaluation"],
+                format=FormatType.Percentage,
+                comparisons=[
+                    DatasetDistributionComparison(
+                        name=class_names[i],
+                        alert=alert_imb[i],
+                        data=[
+                            DatasetDistributionComparisonValue(
+                                value=perc_imb_train[i], alert=alert_imb_train[i]
+                            ),
+                            DatasetDistributionComparisonValue(
+                                value=perc_imb_eval[i], alert=alert_imb_eval[i]
+                            ),
+                        ],
+                    )
+                    for i, _ in enumerate(alert_imb)
```

Comment:
I think the most pythonic way (and supposedly the most efficient way) to index all those lists would be to use `zip`. That being said, it is clearly insignificant here, the other `DatasetWarning`s are not doing that either, and I personally don't particularly like 😝. So up to you, I just thought it was interesting.
```suggestion
                    DatasetDistributionComparison(
                        name=name,
                        alert=alert,
                        data=[
                            DatasetDistributionComparisonValue(value=val_train, alert=alert_train),
                            DatasetDistributionComparisonValue(value=val_eval, alert=alert_eval),
                        ],
                    )
                    for name, alert, val_train, alert_train, val_eval, alert_eval in zip(
                        class_names,
                        alert_imb,
                        perc_imb_train,
                        alert_imb_train,
                        perc_imb_eval,
                        alert_imb_eval,
                    )
```

Reply:
Interesting to know! I don't like it either. I will keep it as is for now, to keep consistency with the others.